Bug: `using` keyword issue bringing module into function scope with global scope throws err

## Minimum code to reproduce:
```odin
import @std
import @arrays

using std, arrays

do main() {
    using std
    println(something)
}

```

## Error:
```shell
error[E3006]: function 'println' found in multiple modules: std, std
  --> examples/example.ez:21:5
   |
21 |     println(addr)
   |     ^ error occurred here
   |
   = help: use explicit module prefix: std.println()

summary: generated 1 error
```

## Issue
It seems that when using the `using` statement globally, rather than taking the global `using` statement and making it the priority the interpreter essentially doubles up on the modules.  Wish I could explain better but Im tired so fuck it.

--- a/pkg/interpreter/object.go
+++ b/pkg/interpreter/object.go
@@ -278,13 +278,25 @@ func (e *Environment) Use(alias string) {
 }
 
 // GetUsing returns all modules in scope (checks parent scopes too)
+// Deduplicates modules to avoid ambiguity when same module is used in multiple scopes
 func (e *Environment) GetUsing() []string {
 	result := make([]string, len(e.using))
 	copy(result, e.using)
 	if e.outer != nil {
 		result = append(result, e.outer.GetUsing()...)
 	}
-	return result
+
+	// Deduplicate the modules
+	seen := make(map[string]bool)
+	deduped := make([]string, 0, len(result))
+	for _, module := range result {
+		if !seen[module] {
+			seen[module] = true
+			deduped = append(deduped, module)
+		}
+	}
+
+	return deduped
 }
 
 func (e *Environment) Get(name string) (Object, bool) {

